### ISSUE TITLE
RELEASE 1.0

### ISSUE THREAD

**Issue by francisco-dlp** [hyperspy/hyperspy#1139]
TODO - RELEASE
--------------------------

* [x] Merge remaining PRs
* [x] Fix unfold
* [x] Merge component std
* [ ] Update demos
* [x] Create RELEASE_0.8.5 branch
    * [x] Update version
    * [x] Update changes
* [x] Create Python installers
* [x] Test Python installers
* [x] Create Windows installers
* [x] Test Windows installers
* [x] Create Release in GitHub (as pre-release)
* [x] Update DOI in ``README.rst`` and ``citing.rst``
* [x] Create Bundle installer
    * [x] Test it with admin rights
    * [x] Test with user without admin rights
* [x] Upload docs
* [x] Upload to pypi
* [x] Upload to GH
* [x] Update website
* [x] Announce in mailing list
* [ ] Backport hdf5 security issue fix

**Comment by sem-geologist** [comment]
the bundle installer can be created in apveyor, also the windows wheels. Unless nsis is still offline, then bad luck... also travis does sdist and osx wheels. all that requires to replace my credentials with authorized ones for hyperspy main

**Comment by sem-geologist** [comment]
... in both: travis and appveyor yml.

**Comment by francisco-dlp** [comment]
"all that requires to replace my credentials with authorized ones for hyperspy main" could you provide details for this. @to266, I bet that we could use your credentials for this, could you submit a PR to change @sem-geologist's to yours?

**Comment by sem-geologist** [comment]
in .travis.yml it is line 47
the new key should be generated in github, then used on linux travis comand line to encrypt, then it lands in that file instead of mine key

for appveoyr it is simplier:
I added my credentials under @to266 and @vidartf , if they are still actual it should be enought just comment out mine, and uncoment one of them. NOTE it is different key than on travis.
However appveyor key is required only for deployment to github, which I couldn't make work (Maybe it will simply work out with @to266  or @vidartf keys, as they probably have better configured it. I dont know. It is note very relevant as after appveoyr builds stuff it preserves it (for ever) as artifacts which can be downloaded.

Travis however needs credentials as it does not preserve those artifacts, and needs to deploy them to github.

**Comment by francisco-dlp** [comment]
ok, thanks!

2016-07-13 17:32 GMT+01:00 Petras Jokubauskas <notifications@github.com>:

**Comment by sem-geologist** [comment]
by the way, it is not called key, but `personal acces token` in github settings. It have to have at least `public_repo` enabled. then that token have to be encrytped with linux travis (probably osx have also comand line tool for that) command line tool like `travis encrypt THE_TOKEN` and result added under `secure:` in travis yml commenting out mine token.

**Comment by francisco-dlp** [comment]
In #1140 I have added a draft of the changes in 1.0. Could you all verify that I didn't miss any contribution that you think that should be highlighted?

**Comment by francisco-dlp** [comment]
Does anybody knows how [to fix Read the Doc](https://readthedocs.org/projects/hyperspy/builds/4195366/)? It is failing since we merged #949 because now it tries to download and compile all the dependencies.

**Comment by sem-geologist** [comment]
I know that mock went away, but this: http://read-the-docs.readthedocs.io/en/latest/faq.html tells to use mock to overcome the issue with compiling stuff while building the docs.

**Comment by francisco-dlp** [comment]
That's clever, thanks! mock didn't go away, it is now in the standard library so there is no need to install it any more.

**Comment by sem-geologist** [comment]
I just pull requested

**Comment by magnunor** [comment]
@francisco-dlp, could maybe include a sentence about new the new notebook demos (https://github.com/hyperspy/hyperspy-demos) in changes.txt

**Comment by francisco-dlp** [comment]
Not sure... that's not part of the release (they don't need to be in sync and, actually, they haven't been updated to 1.0 yet). We have announced them in the website and the mailing list.

**Comment by sem-geologist** [comment]
I am trying to deal with docs, I find it hard to configure mine copy of readthedoc. however I added hacky lines in setup.py which checks setup.py absolute path, and if 'readthedoc.org' is there then makes install_requires an empty list. I want to test it out if it works. Unfortunately this is the price of setuptools. as distutil simply ignored the install_requires (it was always complaining before in logs that it does not understand what is it, now it does not anymore, however we get the concequences with read the doc. Another and maybe more proper way would be to not install hyperspy at readthedoc, but use sphinx-build (I am not sure about this)... I am stuck at 3rd step: 
```
python /home/docs/checkouts/readthedocs.org/user_builds/hyperspy-temp/envs/latest/bin/pip install --exists-action=w --cache-dir /home/docs/checkouts/readthedocs.org/user_builds/hyperspy-temp/.cache/pip -rdocs/pip-req.txt
Could not open requirements file: [Errno 2] No such file or directory: 'docs/pip-req.txt'
```
If that would work it would be great...

**Comment by sem-geologist** [comment]
It works, but not with setup path (I am a bit confused why it didnt worked) but with python executalbe path.

**Comment by sem-geologist** [comment]
this is second try to fix the docs, and that worked: #1155.
And this is evidence: https://readthedocs.org/projects/hyperspy-temp/builds/4199533/
